tiled/src/main/kotlin/ktx/tiled/mapObject.kt

```diff
@@ -0,0 +1,73 @@
+package ktx.tiled
+
+import com.badlogic.gdx.maps.MapObject
+import com.badlogic.gdx.maps.MapProperties
+import com.badlogic.gdx.maps.objects.*
+import com.badlogic.gdx.math.*
+
+/**
+ * Extension method to directly access the [MapProperties] of a [MapObject]. The type is automatically
+ * derived from the type of the given default value. If the property is not defined the defaultValue will be returned.
+ * @param key property name
+ * @param defaultValue default value in case the property is missing
+ * @return value of the property or defaultValue if property is missing
+ */
+inline fun <reified T> MapObject.property(key: String, defaultValue: T): T = this.properties[key, defaultValue, T::class.java]
```

Comment:
Do `MapObject`, `MapLayer` etc. follow any sort of class hierarchy? Maybe the utilities do not have to be added 1 by 1.

Reply:
Unfortunately no. Those are three separate classes that contain a `MapProperties` attribute which is the container for properties.
The OOTB way to access those properties is by calling `.getProperties()` and then call `get`, `contains`, etc..

That's why I have to add them separately.

But maybe it would be a better approach to make them `MapProperties` extensions instead? Then it would be only once and instead of `property` or `containsProperty` you need to write `properties` or `properties.contains`.

What do you think?

Reply:
I would definitely add extensions for `MapProperties`. If you include the `operator` keyword in the `contains` extension method, it will also allow for the `"key" in map.properties` syntax.

As for the individual extensions to extract properties from `MapObject` and so on, I still think they are useful and we can leave them as they are. In other words, all of them could support the following methods:
- `property(key: String): T`: returns a property or throws an exception. Checks for nulls.
- `property(key: String, defaultValue: T): T`: returns a property or the default value. Checks for nulls.
- `propertyOrNull(key: String): T?`: returns a property or null if undefined.
- `containsProperty(key: String): Boolean`: true if property exists and is not null, false otherwise.

I'd leave `contains` without any suffix for checks specific to the map class. E.g. `TiledMap` could use operator `contains(layerName: String): Boolean` to verify if certain layers exist.

Reply:
I tried to add an extension now for MapProperties like:

```kotlin
inline operator fun <reified T> MapProperties.get(key: String): T = get(key, T::class.java)
```

but it is shadowed and therefore ignored. So I think it is not possible to add `MapProperties` extensions?

So for now I will stick to TiledMap, MapObject and MapLayer extensions and will provide the functionality discussed above

Reply:
It would be shadowed only if there is a single parameter `get` method already.

Yeah, well, if there's no way to do it, stick to other extensions. Thanks for checking it out.